### PR TITLE
add gitignore and a function to generate a random slug

### PR DESCRIPTION
**Pull request by jm2242**
#### Pull Request Description
*Short description of the what changes you have made*
Adds a .gitignore and the `generate_random_slug` helper function
#### Requested/Proposed Reviewers
*A short list of people that should probably look at this before it hits dev-main*

#### Known Problems / Yet To Do
*A list of everything that you know might be broken*

#### Things I need Help On:
*A list of things that you would like other people to work with you on to finish the pull request*

### CODE REVIEW

File: walawala/__init__.py [hunk 1]
```diff
@@ -87,6 +89,42 @@ def all_encodings():
...
+    In [8]: generate_random_slug(5)
+    Out[8]: 'GGsqX'
+
+    In [7]: generate_random_slug(15, string.ascii_uppercase)
+    Out[7]: 'MZBZKZMQKEJIESF'
+    """
+    if not isinstance(charset, str):
+        raise TypeError("The provided chararacter set charset must be a string")
+
+    if length < 0:
```
Comment (pelzhaus) [inline walawala/__init__.py]: is 0 length acceptable?

File: walawala/__init__.py [hunk 2]
```diff
@@ -87,6 +89,42 @@ def all_encodings():
...
+    Out[7]: 'AuifexrxQfMx6VJ7JXc9'
+
+    In [8]: generate_random_slug(5)
+    Out[8]: 'GGsqX'
+
+    In [7]: generate_random_slug(15, string.ascii_uppercase)
+    Out[7]: 'MZBZKZMQKEJIESF'
+    """
+    if not isinstance(charset, str):
+        raise TypeError("The provided chararacter set charset must be a string")
```
Comment (pelzhaus) [inline walawala/__init__.py]: are unicodes not allowed?

File: walawala/__init__.py [hunk 3]
```diff
@@ -87,6 +89,42 @@ def all_encodings():
...
+
+    Args:
+       length (int): A positive integer length of the outputed random slug
+       charset (str): The characters to choose from when generated the random slug
+
+    Returns:
+       string: the randomly generated slug
+
+    Raises:
+        TypeError: if the provided charset is not a string
```
Comment (pelzhaus) [inline walawala/__init__.py]: or assertion error!

File: walawala/__init__.py (hunk 1 above)
Comment (jm2242) [reply walawala/__init__.py]: it'll return an empty string (actually so will negatives) so it seems ok to leave it

File: walawala/__init__.py (hunk 2 above)
Comment (jm2242) [reply walawala/__init__.py]: i should probably allow that

**Comment by jm2242** [comment]
if I used `uuid`, then I would have to write a helper to convert that to the slug I want, right? In which case, I might as well write a helper anyone can use? @kqdtran

**Comment by jm2242** [comment]
We are guaranteed randomness to the extent that `random.SystemRandom()` guarantees randomness

**Comment by kqdtran** [comment]
It literally is just a call to `import uuid` and `random_slug = str(uuid.uuid4())` -- that's so much cleaner and you can then index that to get however many characters you want. I still don't get why this helper is useful when there is already a built-in way of doing it. To me, it's almost like writing your own `string.split()` function when it already came with stdlib. Lemme know if I'm misunderstanding something.

**Comment by jm2242** [comment]
@jakeseaton go do your homework

**Pull request merged**